dips/views.py

```diff
@@ -1,72 +1,55 @@
 from django.forms import modelform_factory
 from django.http import Http404, HttpResponse
 from django.shortcuts import render, redirect, get_object_or_404
 from django.utils.translation import gettext as _
+from .helpers import get_sort_params, get_page_from_search
 from .models import User, Collection, DIP, DigitalFile, DublinCore
-from .forms import DeleteByDublinCoreForm, UserCreationForm, UserChangeForm, DublinCoreSettingsForm
+from .forms import (DeleteByDublinCoreForm, UserCreationForm, UserChangeForm,
+                    DublinCoreSettingsForm)
 from .tasks import extract_and_parse_mets
+from search.functions import (add_query_to_search, add_digital_file_aggs,
+                              add_digital_file_filters)
 
 
-def get_sort_params(request, options, default):
+def _get_and_validate_digital_file_filters(request):
     """
-    Get sort option and direction from request. Check options dict. for
-    available choices and use default if no option is passed on the request
-    or if the option is not valid. Defaults to asc. sort direction.
+    Obtains the digital file filters from the request GET parameteres and
+    validates the start and end dates. Returns two dics, the first one with
+    all the filters set (to maintain their value in the templates) and the
+    second one with only the valid filters (to be applied to the search).
+    Adds error messages to the request for invalid dates.
     """
-    option = default
-    if 'sort' in request.GET and request.GET['sort'] in list(options.keys()):
-        option = request.GET['sort']
-
-    direction = 'asc'
-    if 'sort_dir' in request.GET and request.GET['sort_dir'] in ['asc', 'desc']:
-        direction = request.GET['sort_dir']
-
-    return (option, direction)
-
-
-def add_query_to_search(search, request, fields):
-    """
-    Check if a `query` parameter has been sent in the request and add a
-    `simple_query_string` to the search over a given list of fields.
-    """
-    if 'query' in request.GET and request.GET['query']:
-        search = search.query(
-            'simple_query_string',
-            query=request.GET['query'],
-            default_operator='and',
-            fields=fields,
-        )
-
-    return search
-
-
-def get_page_from_search(search, request):
-    """
-    Create paginator and return current page based on the current search
-    and the page and limit request parameters. Limit defaults to 10 and
-    can't be set over 100.
-    """
-    try:
-        limit = int(request.GET.get('limit', 10))
-        if limit <= 0 or limit > 100:
-            raise ValueError
-    except ValueError:
-        limit = 10
-
-    paginator = Paginator(search, limit)
-    page_no = request.GET.get('page')
-    try:
-        page = paginator.page(page_no)
-    except PageNotAnInteger:
-        page = paginator.page(1)
-    except EmptyPage:
-        page = paginator.page(paginator.num_pages)
+    filters = {
```

Comment:
The usual "Django" way to handle filters/validation like this is usually a Form class — even if you don't display the form as HTML it's a good way to encapsulate a collection of fields with validation and errors. That's probably too big a change for this PR though.

Reply:
I struggled a lot with this @cole, I initially wanted to use a proper Django form but I thought it was going to be too complex to build and validate considering how the digital file filters work. Getting the data from Elasticsearch aggregations and building multi-selects with check-boxes. I'm not saying it's impossible but, as you mention, it's quite a big change at this point.

Reply:
@jraddaoui agreed, it's a bit of a pain to get working. Just making sure you'd thought about that approach. Definitely not for this PR.